src/CMakeLists.txt

```diff
@@ -54,8 +53,8 @@ set_target_properties(libappimageinfo PROPERTIES
 add_dependencies(libappimageinfo libappimage libappimageinfo)
 
 add_executable(appimageinfotool app/main.cpp)
-target_link_libraries(appimageinfotool libappimageinfo ${APPIMAGEKIT_LIBRARIES} ${LIBBFD_LIBRARIES})
-target_include_directories(appimageinfotool PRIVATE ${NLOHMANN_JSON_INCLUDE_DIRS})
+target_link_libraries(appimageinfotool libappimage libappimageinfo ${APPIMAGEKIT_LIBRARIES} ${LIBBFD_LIBRARIES})
```

Comment:
`libappimage` is a dependency of `libappimageinfo`, therefore CMake will automatically link to it here. This should not be necessary and is an anti-pattern. If it doesn't work, you need to fix the "scoping" of the libraries you link to in line 28ff. I can help with that if you want to.